pipeline/metadata/add_metadata.py

```diff
@@ -0,0 +1,96 @@
+"""Class for adding ip metadata to rows"""
+
+from __future__ import absolute_import
+
+import datetime
+from typing import List, Tuple, Iterator, Iterable
+
+import apache_beam as beam
+
+from pipeline.metadata.beam_metadata import DateIpKey, IP_METADATA_PCOLLECTION_NAME, ROWS_PCOLLECION_NAME, make_date_ip_key, merge_metadata_with_rows
+from pipeline.metadata.schema import BigqueryRow, IpMetadataWithKeys
+from pipeline.metadata.ip_metadata_chooser import IpMetadataChooserFactory
+
+
+class MetadataAdder():
+  """This class exists to provide access to the instantiated IpMetadata inside a beam operation."""
+
+  def __init__(self,
+               metadata_chooser_factory: IpMetadataChooserFactory) -> None:
+    self.metadata_chooser_factory = metadata_chooser_factory
+
+  def add_metadata(
```

Comment:
Perhaps be more explicit with something like `annotate_row_ips`

Reply:
done